File: Habitica/src/com/magicmicky/habitrpgwrapper/lib/models/SuppressedModals.java
package com.magicmicky.habitrpgwrapper.lib.models;

import com.habitrpg.android.habitica.HabitDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

/**
 * Created by viirus on 15/11/15.
 */
@Table(databaseName = HabitDatabase.NAME)
public class SuppressedModals extends BaseModel {

    @Column
    @PrimaryKey
    public String userId;

    private Boolean streak;
    private Boolean raisePet;
    private Boolean hatchPet;
    private Boolean levelUp;

    public Boolean getStreak() {return streak; }
    public Boolean getRaisePet() {return raisePet; }
    public Boolean getHatchPet() {return hatchPet; }
    public Boolean getLevelUp() {return levelUp; }

    public void setStreak(Boolean streak) {this.streak = streak; }
    public void setRaisePet(Boolean raisePet) {this.raisePet = raisePet; }
    public void setHatchPet(Boolean hatchPet) {this.hatchPet = hatchPet; }
    public void setLevelUp(Boolean levelUp) {this.levelUp = levelUp; }
}

File: shared/src/androidMain/kotlin/com/habitrpg/shared/habitica/models/user/SuppressedModals.kt
package com.habitrpg.shared.habitica.models.user

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

actual open class SuppressedModals: RealmObject() {
    @PrimaryKey
    actual var userId: String? = null

    internal actual var preferences: Preferences? = null
    actual var streak: Boolean? = null
    actual var raisePet: Boolean? = null
    actual var hatchPet: Boolean? = null
    actual var levelUp: Boolean? = null
}